fix: Be able to load meteor apps from Shopware bundles

JoshuaBehrens opened the pull request:
### 1. Why is this change necessary?

There has been a trend towards providing changes by bundle instead of plugin due to the added complexity of the plugin management, that does not need to be managed if not used. To still provide an administration experience with new Meteor SDK you need to be a plugin.

To be able to only ship in bundles in Shopware 6.6 you have to have a Shopware bundle with an additional bundle, that is technically a plugin and ship it like this:

```php
final class InterimBundle extends Bundle
{
    #[\Override]
    public function getAdditionalBundles(AdditionalBundleParameters $parameters): array
    {
        $projectDir = $parameters->getKernelParameters()['kernel.project_dir'];
        $pluginDir = __DIR__ . DIRECTORY_SEPARATOR . 'RealBundle . DIRECTORY_SEPARATOR;

        return [
            new BundleDisguisedAsPlugin(true, \mb_substr($pluginDir, \mb_strlen($projectDir) + 1), $projectDir),
        ];
    }
}
```

When you want to have a working suffix you also need to have a patch in the webpack.config to remove the bundle suffix as well but this is not practical as shopware-cli always pulls a fresh Shopware installation out of the hat. So you better do not name your bundle like Symfony tells you to do.

So please just let us clean this up for Shopware 6.7 to not be an issue anymore :) 

### 2. What does this change do, exactly?

It moves a method to a base classes and change one check from plugin to bundle class.

It adds one bundle suffix removal as this is very Symfony-esque to do that:

* Create bundle with suffix bundle
* Remove bundle suffix everywhere else e.g. find extension class and assets path

### 3. Describe each step to reproduce the issue or behaviour.

1. Folow the tutorial to start with meteor
2. Have a bundle with a with a file `Resources/app/administration/index.html` and `Resources/app/administration/src/main.js`
3. Compile the admin
4. Install assets
5. Have no  ![grafik](https://github.com/user-attachments/assets/8e92877a-3109-4bb4-8dc3-63c8df3b5ce6)
6. but rather a 404

### 4. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/shopware/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

JoshuaBehrens committed:
Be able to load meteor apps from Shopware bundles

JoshuaBehrens committed:
Symfony bundle suffix hits once more

JoshuaBehrens commented:
https://github.com/shopware/shopware/issues/10753 here we have an idea how to approach the issues in Shopware 6.6 I mentioned in

> When you want to have a working suffix you also need to have a patch in the webpack.config to remove the bundle suffix as well but this is not practical as shopware-cli always pulls a fresh Shopware installation out of the hat. So you better do not name your bundle like Symfony tells you to do.

mitelg commented on changelog/_unreleased/2025-06-13-allow-admin-scripts-provided-by-bundles.md:
```diff
@@ -0,0 +1,9 @@
+---
+# Core
+* Changed focus in administration asset path detection from `\Shopware\Core\Framework\Plugin` to `\Shopware\Core\Framework\Bundle` to be able to load meteor apps from bundles  
+* Fixed usage of superfluous bundle suffix in administration asset path building in `\Shopware\Core\Framework\Api\Controller\InfoController`
```
```suggestion
* Changed bundle suffix in administration asset path building in `\Shopware\Core\Framework\Api\Controller\InfoController` by removing bundle suffix
```

mitelg commented on tests/integration/Core/Framework/Api/Controller/InfoControllerTest.php:
```diff
@@ -498,6 +501,17 @@ public function testBaseAdminPaths(): void
         $config = json_decode($content, true, 512, \JSON_THROW_ON_ERROR);
         static::assertCount(3, $config['bundles']);
```
```suggestion
        static::assertCount(5, $config['bundles']);
```

as you added two more bundles, this should be five, right?  

nevertheless you need to check this test, as it is failing

JoshuaBehrens replied:
Thank you for checking back. Reasonable question. It should be 4 as InfoController::getBundles filters one out with this check:
```php
            if (empty($styles) && empty($scripts) && $baseUrl === null) {
                continue;
            }
```

JoshuaBehrens commented:
I rebased, squashed and followed your and the CI suggestions.

I find it a bit sad, that I had to reduce this as Plugin is referencing a class not a namespace ...
```diff
-                     * @see \Shopware\Core\Framework\Plugin\Util\AssetService::getTargetDirectory
+                     * @see Plugin\Util\AssetService::getTargetDirectory
```

mitelg commented:
thanks for your contribution @JoshuaBehrens 👍🎉💙

JoshuaBehrens commented:
Thank you for merging :)
